filesystem/compatibility.go

```diff
@@ -0,0 +1,116 @@
+package filesystem
+
+import (
+	"io/fs"
+	"os"
+	"path"
+	"time"
+)
+
+type fsCompatible struct {
+	fs FileSystem
+}
+
+type fsFileWrapper struct {
+	File
+	stat os.FileInfo
```

Comment:
I've been using this with read-only images. But if this is to support read/write, then caching the `stat` info is probably a mistake.

Reply:
It isn't. `fs.FS` currently is read-only. Even its few sub-variants are read-only. If it helps, I have implemented a full read-write FS variant (`fs.FS` compatible) which includes support for case-sensitivity and hardlinks and chmod and chown, even on underlying filesystems that do not support it. If you want a link, just ask.

Reply:
Right, fs.FS is read-only, but `diskfs.FileSystem` can be read/write, so this abstraction could break?

Reply:
I don't see how. `fs.FS` is just a subset of what is here. I am comfortable with what it is.